lua/precognition/ffi.lua

```diff
@@ -0,0 +1,29 @@
+local M = {}
+
+---@return ffi.namespace*
+function M.load()
+    if not _G.precog_C then
+        local ffi = require("ffi")
```

Comment:
Imo this shouldn't need a pcall. If it's throwing an error, this is a serious enough implementation detail that we should make sure to solve the actual problem.

Reply:
I borrowed the implementation from Noice.

Folke has a note there that

```
-- Put in a global var to make sure we dont reload
  -- when plugin reloaders do their thing
```